docs/DeveloperGuide.md

```diff
@@ -128,14 +157,54 @@ The `Storage` component,
 
 ### Common classes
 
-Classes used by multiple components are in the `seedu.mcgymmy.commons` package.
+Classes used by multiple components are in the `jimmy.mcgymmy.commons` package.
 
 --------------------------------------------------------------------------------------------------------------------
 
 ## **Implementation**
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Import feature
+
+The current Import feature is facilitated by `JsonMcGymmyStorage`. 
+It extends model with the ability to override the current data with an imported one. 
+
+#### Implementation
+
+Given below is an example usage scenario and how the import mechanism behaves at each step
+
+Step 1. The User just installed his application on his new computer and wants to transfer his old data over
+
+Step 2. The user transfers his previous save file to `C:/McGymmy/saveFile.json`
+
+Step 3. The User executes `import c:/McGymmy/saveFile.json`. 
+The import command will check if the file is valid and exists before calling `JsonMcGymmyStorage`. 
+`JsonMcGymmyStorage` will call `#readMcGymmy` if the read is successful, the old data will be overwritten.
+Otherwise, a CommandException will be thrown.
+
+Step 4. His old data files from his old computer will be overridden by his old data
+
+The following sequence diagram shows how the import operation works:
+
+![ImportSequenceDiagram](images/ImportSequenceDiagram.png)
```

Comment:
I think we can shave off the parsing part bc it would have been mentioned in the above sections.
Makes the diagram less packed

Reply:
Ookie gimmie a sec